### PR TITLE
doc: initialize a user-facing mdbook

### PR DESCRIPTION
**Pull request by RaitoBezarius**
@sonmarcho @R1kM Here's my proposal regarding documenting `progress` among other things that I learned during the FV of AVL trees.

A mdBook-style user facing documentation with a part on "Proving Rust code with Lean" which dumps all that I know about using Aeneas & Lean in practice on Rust code.

Question is: do I do everything in this PR or shall we just get the CI right and improve on it incrementally as we commit to an outline, etc. ?

### CODE REVIEW

**Comment by sonmarcho** [comment]
Thanks for starting this, it was much needed!
Following our conversation, we should first get the CI right and do things incrementally. In particular, once we get the outline and the CI working, anybody can start making changes.

**Comment by RaitoBezarius** [comment]
Moved to an Alectryon setup.

**Comment by sonmarcho** [comment]
@Nadrieril I think it's important that you keep an eye on this PR as well.

**Comment by RaitoBezarius** [comment]
@Nadrieril ping on this :)

File: .github/workflows/build-userdocs.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Test Aeneas user docs
+    runs-on: [self-hosted, linux, nix]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```
Comment (Nadrieril) [inline .github/workflows/build-userdocs.yml:11]: Why this? By default the clone is shallow, this makes it deep. I don't think we need that

**Review by Nadrieril — APPROVED** [review]

File: .github/workflows/build-userdocs.yml (hunk 1 above)
Comment (RaitoBezarius) [reply .github/workflows/build-userdocs.yml:11]: for building, yes, i'm dumb

Comment (RaitoBezarius) [reply .github/workflows/build-userdocs.yml]: Fixed.

**Comment by sonmarcho** [comment]
The PR https://github.com/AeneasVerif/aeneas/pull/282 is blocked because the doc building fails.

**Comment by RaitoBezarius** [comment]
Urgh.

**Comment by RaitoBezarius** [comment]
Sent a hotfix in another PR.

**Pull request merged**